File: baekjoon/src/stack/No10773.java
package stack;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

/**
 * https://www.acmicpc.net/problem/10773
 */
public class No10773 {
    private static final Stack<Integer> stack = new Stack<>();

    public static void command(int number) {
        if (number == 0 && !stack.empty())
            stack.pop();
        else
            stack.push(number);
    }

    public static int getSumOfStack() {
        int sum = 0;
        while (!stack.empty())
            sum += stack.pop();
        return sum;
    }

    public static void main(String[] args) {
        try (BufferedReader reader = new BufferedReader(new InputStreamReader(System.in))) {
            int numberOfCommands = Integer.parseInt(reader.readLine());
            for (int i = 0; i < numberOfCommands; i++)
                command(Integer.parseInt(reader.readLine()));
            System.out.println(getSumOfStack());
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: baekjoon/src/tags/dataStructure/No10773.kt
package tags.dataStructure

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.*

/**
 * https://www.acmicpc.net/problem/10773
 */
fun main() {
    val reader = BufferedReader(InputStreamReader(System.`in`))
    val writer = BufferedWriter(OutputStreamWriter(System.out))
    val stack = Stack<Int>()

    val numberOfInput = reader.readLine().toInt()
    repeat(numberOfInput) {
        val number = reader.readLine().toInt()
        if (number == 0)
            stack.pop()
        else
            stack.push(number)
    }

    var sum = 0
    while (!stack.isEmpty())
        sum += stack.pop()
    writer.write("$sum")
    writer.flush()
}
